Vizzuality/landgriffon
# Feature: LANDGRIF-874 update smart filters

### General description

Update tree entity smart filters to conditionally accept a scenario Id to then retrieve elements that are present in sourcing locations and their ancestry from the main table.

Refactors get<EnitityName>Ancestry from each repository class to a main method of the extended ExtendedTreeRepository, and improves some readability

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

Create a Scenario with intervention(s) which replaces Materials, AdminRegions, Suppliers and BusinessUnits existing in Sourcing Locations. Then query all dropdowns, including the scenarioId. you should now see these new elements in the tree

- Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging
